lib/Analysis/Allocation.cpp

```diff
@@ -126,6 +125,12 @@ SmallVector<unsigned> getScratchConfigForAtomicRMW(triton::AtomicRMWOp op) {
   return smemShape;
 }
 
+SmallVector<unsigned> getScratchConfigForAtomicCAS(triton::AtomicCASOp op) {
+  SmallVector<unsigned> smemShape;
+  smemShape.push_back(1);
+  return smemShape;
```

Comment:
nit: 
return SmallVector<unsigned>{1} 
should be good enough.

BTW, @ptillet it seems that the legacy code only implements scalar type for AtomicCAS, but i didn't see such information in the official document of triton. So is this just a TODO? or some other considerations

Reply:
Done